docs/proposals/mvp-dashboard-design.md

```diff
@@ -43,18 +43,82 @@ can query the relevant test builds.
 
 ## Proposal
 
-![Diagram 1](../images/perf-dash-proposal-diagram.png)
+![alt text](../images/perf-dash-proposal-diagram.png "Perf-Dash Diagram")
+
+### Server API
+
+The system we propose requires the development of a server API that will be
+used to serve queries that a user makes via the dashboard. To speed up the
+initial development process, these queries will be hardcoded for test results.
+Adding query build functionality for the user is a feature that will be
+developed in future development cycles.
+
+The server polls the data storage unit for performance test logs. The frequency
+of the polling will be adjusted after implementation so that it reflects the
+rate that test results enter the data storage. The server is to maintain a
+cache of frequently and recently accessed data. The server expects a json
+result from the data store.
+
+This data can then be received by the dashboard.
+
+### Dashboard
+
+The dashboard serves the purpose of allowing the user to make queries of the
+results stored in the data storage unit. The dashboard provides a simple user
+interface with a list of drop down menus for options such as: lab, config,
+test, run, and metric. The dashboard communicates with the server (both
+running on the same node) using an NGINX reverse proxy.
+
+The query is made via HTML requests in the form of `GET /api/labs`,
+`GET /api/labs/{lab id}`. The expected result is a json with the appropriate
+test data. This json is then used to create the graph of the specified metrics
+as they trend over the past 50 builds.
+
+Our choice of developing a REST API provides the benefit that any graph
+configured by a user can be sured via the URL link to that page.  This allows
+for the fast sharing of performance test results between team members and
+other teams.
```

Comment:
Shareable links and the decision to use a RESTful API are not related. The implementation for 'shareable' links will probably come from the Angular application itself, not the backend API.

I don't think we need to argue why we choose a RESTful API. The obvious benefits, it's that you have a machine-consumable interface to easily build other tools and applications on top of the API, and it abstracts the storage implementation so user don't directly depend on our choice of data storage system.